runtimes/core/src/api/manager.rs

```diff
@@ -337,7 +335,7 @@ impl Manager {
         let server = HttpServer::new(encore_routes, api, fallback);
 
         let api_listener = self.api_listener.lock().unwrap().take();
-        let gateway_listener = self.gateway_listener.lock().unwrap().take();
+        let gateway_listener = self.gateway_listener.clone();
```

Comment:
This can be `.take()` instead

Reply:
Then I would have to put it back into a mutex. I can do that if you prefer

Reply:
or I can just do .as_ref actually!

Reply:
Ah fair, can leave it as-is then. It was the case before that we would use `None` to detect that the server had already been started, but that's no longer the case so not much point in using `take()`

Reply:
as_ref actually didnt work here, due to lifetimes (since we borrow self)